File: backend/utils/roles.js
const VALID_ROLES = ["admin", "seller", "supplier", "user"];

export const getUserRole = (user) => {
  if (!user) return "user";

  const nestedRole = user.get?.("user.role") || user.user?.role;
  const topLevelRole = user.get?.("role") || user.role;

  if (VALID_ROLES.includes(nestedRole)) return nestedRole;
  if (VALID_ROLES.includes(topLevelRole)) return topLevelRole;

  return "user";
};

File: backend/app/utils/roles.py
VALID_ROLES = ["admin", "seller", "supplier", "user"]

def get_user_role(user) -> str:
    if not user:
        return "user"
    
    # Check top-level attribute
    role = getattr(user, "role", "user")
    if role in VALID_ROLES:
        return role
        
    return "user"
